### PR TITLE
chore: Build with Typescript rather than Babel

### PR DESCRIPTION
**Pull request by samuelfreiberg**
This improvement stems from [Issue #18739](https://github.com/appium/appium/issues/18739).

This package is bringing in a locked version of @babel/runtime to all consumers. This is bad for a couple major reasons:

1. Teams working in a JS Monorepo cannot use yarn to standardize on a single version.
2. Doesn't allow for auto-update tools (like GitHub's dependabot) and internal tools of large companies to automatically deal with version updates.

This PR moves from building with Babel to Typescript. Also removes the dependency on Babel so the issue for consumers no longer persists.

### CODE REVIEW

**Comment by samuelfreiberg** [comment]
@jlipps @boneskull @mykola-mokhnach In regards to [Issue #18739](https://github.com/appium/appium/issues/18739)

**Comment by mykola-mokhnach** [comment]
Thanks @samuelfreiberg 
Please resolve conflicts and rename the PR according to conventional commits requirements

**Comment by samuelfreiberg** [comment]
> Thanks @samuelfreiberg Please resolve conflicts and rename the PR according to conventional commits requirements

Done

**Comment by samuelfreiberg** [comment]
@mykola-mokhnach Question - Assuming this PR gets approved, what's the timeline for it to make a new version, and for Appium and all dub-dependencies to take that version?

I'm asking because we're working on a critical project dependent on this, and it would help us plan accordingly knowing a rough timeline.

Thank you!

**Comment by jlipps** [comment]
OK, I'm running CI to see how tests fare. Code looks good so far! Once merged, I don't mind making a new release of Appium 2 in the next day or two, which should resolve the issue.

**Comment by jlipps** [comment]
yeah, looks like your npm registry was set to `pkgs.dev.azure.com`. should just be the default npm registry.

File: index.js [hunk 1]
```diff
@@ -1 +1,12 @@
-module.exports = require('./build/lib/index.js');
+import {install} from 'source-map-support';
```
Comment (boneskull) [inline index.js:1]: why did this file move?

**Review by boneskull — APPROVED** [review]
thanks.  I think this is as close to perfect as I could expect.  there are some formatting issues but it doesn't look like we're running `prettier` as a pre-commit hook, so that's fine for now.

**Comment by boneskull** [comment]
once `package-lock.json` is resolved, we can squash and release this

**Comment by samuelfreiberg** [comment]
Thanks for the input @boneskull and @jlipps. I have fixed the registry issue with the lock file

**Comment by boneskull** [comment]
@samuelfreiberg out of curiosity, are you using this package w/appium, or just as a general-purpose process spawner?

**Review by jlipps — APPROVED** [review]

**Comment by jlipps** [comment]
ok this has been published as `2.0.3`. we should see it make its way into appium in the next day or two

**Comment by mykola-mokhnach** [comment]
@samuelfreiberg I believe something is still not right there. Could you please check https://github.com/appium/appium/issues/18756 ?

**Comment by samuelfreiberg** [comment]
I believe I found a solution - working on getting a PR out

**Pull request merged**